internal/authz/authz_test.go

```diff
@@ -179,3 +206,136 @@ func TestIndividualAuthorize(t *testing.T) {
+							DirectlyRelatedUserTypes: []*openfgav1.RelationReference{
+								{Type: "user"},
+							},
+						},
+					},
+				},
+			},
+			{
+				Type: "folder-with-module",
+				Relations: map[string]*openfgav1.Userset{
+					"viewer": typesystem.This(),
+				},
+				Metadata: &openfgav1.Metadata{
+					Module: module1,
+					Relations: map[string]*openfgav1.RelationMetadata{
+						"parent": {
+							DirectlyRelatedUserTypes: []*openfgav1.RelationReference{
+								{Type: "folder"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	typesys := typesystem.New(model)
+
+	t.Run("error_when_write_tuples_errors", func(t *testing.T) {
+		modules, err := authorizer.GetModulesForWriteRequest(
+			&openfgav1.WriteRequest{
+				StoreId: "store-id",
+				Writes: &openfgav1.WriteRequestWrites{
+					TupleKeys: []*openfgav1.TupleKey{
+						{Object: "unknown:2", Relation: "viewer", User: "user:jon"},
+					},
+				},
+			},
+			typesys,
+		)
+		require.Error(t, err)
+		require.Empty(t, modules)
+	})
+
+	t.Run("error_when_delete_tuples_errors", func(t *testing.T) {
+		modules, err := authorizer.GetModulesForWriteRequest(
+			&openfgav1.WriteRequest{
+				StoreId: "store-id",
+				Deletes: &openfgav1.WriteRequestDeletes{
+					TupleKeys: []*openfgav1.TupleKeyWithoutCondition{
+						{Object: "unknown:2", Relation: "viewer", User: "user:jon"},
+					},
+				},
+			},
+			typesys,
+		)
+		require.Error(t, err)
+		require.Empty(t, modules)
+	})
+
+	t.Run("return_empty_when_a_write_tuple_has_no_modules", func(t *testing.T) {
+		modules, err := authorizer.GetModulesForWriteRequest(
+			&openfgav1.WriteRequest{
+				StoreId: "store-id",
+				Writes: &openfgav1.WriteRequestWrites{
+					TupleKeys: []*openfgav1.TupleKey{
+						{Object: "folder-with-module:2", Relation: "viewer", User: "user:jon"},
+						{Object: "folder:2", Relation: "viewer", User: "user:jon"},
+					},
+				},
+			},
+			typesys,
+		)
+		require.NoError(t, err)
+		require.Empty(t, modules)
+	})
+
+	t.Run("return_empty_when_a_delete_tuple_has_no_modules", func(t *testing.T) {
+		modules, err := authorizer.GetModulesForWriteRequest(
+			&openfgav1.WriteRequest{
+				StoreId: "store-id",
+				Deletes: &openfgav1.WriteRequestDeletes{
+					TupleKeys: []*openfgav1.TupleKeyWithoutCondition{
+						{Object: "folder-with-module:2", Relation: "viewer", User: "user:jon"},
+						{Object: "folder:2", Relation: "viewer", User: "user:jon"},
+					},
+				},
+			},
+			typesys,
+		)
+		require.NoError(t, err)
+		require.Empty(t, modules)
+	})
+
+	t.Run("return_modules", func(t *testing.T) {
+		modules, err := authorizer.GetModulesForWriteRequest(
+			&openfgav1.WriteRequest{
+				StoreId: "store-id",
+				Deletes: &openfgav1.WriteRequestDeletes{
+					TupleKeys: []*openfgav1.TupleKeyWithoutCondition{
+						{Object: "folder-with-module:2", Relation: "viewer", User: "user:jon"},
+					},
+				},
+			},
+			typesys,
+		)
+		require.NoError(t, err)
+		require.Equal(t, []string{module1}, modules)
+	})
```

Comment:
Do we need the case where return modules for write?  We only test the case for delete